SBM implementation in the new version of bgms

sekulovskin opened the pull request:
Additionally, I also tried fixing the saved prior inclusion probability in the arguments list of the output.

MaartenMarsman commented on R/function_input_utils.R:
```diff
@@ -207,14 +208,14 @@ check_model = function(x,
-      if(!is.finite(beta_bernoulli_alpha) || !is.finite(beta_bernoulli_beta) || !is.finite(dirichlet_alpha))
-        stop("The scale parameters of the beta and Dirichlet distribution need to be finite.")
+      if(!is.finite(beta_bernoulli_alpha) || !is.finite(beta_bernoulli_beta) || !is.finite(dirichlet_alpha) || !is.finite(lambda))
+        stop("The scale parameters of the beta distrubution, the concentration parameter of the Dirichlet distribution and the rate parameter of the Poisson distribution need to be finite.")
```
"distrubution" -> "distribution"

MaartenMarsman commented on R/bgm.R:
```diff
@@ -158,13 +163,20 @@
+#'   \itemize{
+#'     \item A vector \code{allocations} with the estimated cluster assignments
+#'     of the nodes, calculated using a method proposed by \code{Dahl (2009)} and
+#'     also used by \code{Geng et al. (2019)}.
```
Why did you not use insertCite here?

MaartenMarsman commented on R/bgm.R:
```diff
@@ -435,6 +449,14 @@ bgm = function(x,
+  } else {
+    inclusion_probability = beta_bernoulli_alpha / (beta_bernoulli_alpha + beta_bernoulli_beta)
+  }
+
```
This will break the rcpp function, which expects a matrix, not a scalar, as input for inclusion_probability. Please revert this and the edit on line 472. Can you open a GH issue for fixing the prior inclusion probability in arguments?

MaartenMarsman commented on R/bgm.R:
```diff
@@ -524,13 +548,15 @@ bgm = function(x,
                       allocations = summarySbm$allocations,
-                      clusters = summarySbm$no_clusters,
                       arguments = arguments)
       } else {
```
I need to check if components has names.

sekulovskin replied:
![1dfd1393b7b4d8b23d32a7029e67c289](https://github.com/user-attachments/assets/618d9d1e-8d7d-47d8-8d93-f06489b285ce)
This is how it looks! In easybgm it is printed in a more user-friendly way (as I showed you).

MaartenMarsman commented on R/function_input_utils.R:
```diff
@@ -207,14 +208,14 @@ check_model = function(x,
-        stop("Values for both scale parameters of the beta and Dirichlet distribution need to be specified.")
+         is.null(dirichlet_alpha) || is.null(dirichlet_alpha) || is.null(lambda) || is.null(lambda))
+        stop("Values for both scale parameters of the betathe concentration parameter of the Dirichlet distribution and the rate parameter of the Poisson distribution need to be specified.")
     }
```
betathe --> beta, the 
and the rate --> , and the rate

MaartenMarsman commented on R/bgm.R:
```diff
@@ -524,13 +548,15 @@ bgm = function(x,
+        summarySbm <- summary_SBM(cluster_allocations = out$allocations,
+                                  dirichlet_alpha = dirichlet_alpha,
+                                  lambda = lambda)
 
```
"<-" --> "="

MaartenMarsman commented on R/posterior_utils.R:
```diff
@@ -36,19 +36,115 @@ getDahl = function(cluster_allocations) {
-  # Compute the number of unique clusters for each iteration
-  clusters = apply(cluster_allocations, 1, function(row) length(unique(row)))
+compute_p_k_given_t <- function(t,
+                                log_Vn,
```
"<-" --> "="

MaartenMarsman commented on R/posterior_utils.R:
```diff
@@ -36,19 +36,115 @@ getDahl = function(cluster_allocations) {
+                                no_variables,
+                                lambda) {
+  # Define the K_values
+  K_values <- as.numeric(1:no_variables)
```
Here and below: "<-" --> "=".

MaartenMarsman reviewed (changes requested):
Looking good Nikola. Can you make the suggested style edits, and revert the setting of the inclusion_probability? Then, I can merge it with the main branch.

sekulovskin commented:
> Looking good Nikola. Can you make the suggested style edits, and revert the setting of the inclusion_probability? Then, I can merge it with the main branch.

Hi Maarten,

I think I have addressed all your comments.
